src/main/java/seedu/weeblingo/logic/commands/LearnCommand.java

```diff
@@ -21,6 +21,7 @@
     public CommandResult execute(Model model) {
         requireNonNull(model);
         model.updateFilteredFlashcardList(PREDICATE_SHOW_ALL_FLASHCARDS);
+        model.getMode().switchModeLearn();
```

Comment:
This might be violating Law of Demeter, could there be better way to abstract these type of statements?

Reply:
I think so too. However only the model is passed into execute and since the mode resides in modelmanager I cant find a way to abstract it out

Reply:
What about creating methods to switch to learn mode inside the model manager?

Reply:
good idea, implemented